src/vanilla/utils/proxyMap-indexMap.ts

```diff
@@ -0,0 +1,154 @@
+import { getVersion, proxy } from '../../vanilla.ts'
+
+const maybeProxify = (x: any) => proxy({ x }).x
+
+const isProxy = (x: any) => getVersion(x) !== undefined
```

Comment:
So, this should improve
```suggestion
import { getVersion, proxy, unstable_getInternalStates } from '../../vanilla.ts'

const { proxyStateMap } = unstable_getInternalStates();

const maybeProxify = (x: any) => typeof x === 'object' ? proxy({ x }).x : x

const isProxy = (x: any) => proxyStateMap.has(x)
```